### PR TITLE
Improve AnyError support in `Result.init(attempt:)`

### PR DESCRIPTION
**Pull request by ikesyo**
Originally reported in https://github.com/mdiep/Tentacle/issues/66.

### CODE REVIEW

File: Result/Result.swift [hunk 1]
```diff
@@ -31,7 +31,10 @@ public enum Result<T, Error: Swift.Error>: ResultProtocol, CustomStringConvertib
 	public init(attempt f: () throws -> T) {
 		do {
 			self = .success(try f())
-		} catch {
+		} catch var error {
+			if Error.self == AnyError.self {
```
Comment (mdiep) [inline Result/Result.swift]: This should probably only be if `error` _isn't_ an `AnyError`? (Can you add a test for that?)

Comment (ikesyo) [inline Result/Result.swift]: I think that is already covered here: https://github.com/antitypical/Result/blob/65d0c81e295e3781c9fea5bec309be049b045a3a/Result/Result.swift#L183-L187

Comment (mdiep) [inline Result/Result.swift]: 👍 

I didn't realize that.

**Comment by ikesyo** [comment]
Thanks for the review! 🚢

**Pull request merged**